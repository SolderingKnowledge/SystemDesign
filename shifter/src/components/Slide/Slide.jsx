export default function Slide( props ) {
    return (
        <li className="slide" tabIndex="0">
            <h1 className="number-item" aria-label={`slide ${props.slide}`}>{props.slide}</h1>
        </li>
    );
}
