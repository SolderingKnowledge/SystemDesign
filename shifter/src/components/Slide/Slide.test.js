import Enzyme, { shallow, toJson, mount, render } from "enzyme";
import React from "react";
import Slide from "./Slide";

describe("Slide Component should render", () => {
    const mockItem = 1;
    it("should show text and match snapshot", ()=>{

        const component = shallow(<Slide slide={mockItem} />);
        const text = component.find("h1");
        expect(text.text()).toBe("1");
        expect(component).toMatchSnapshot();
    })
});