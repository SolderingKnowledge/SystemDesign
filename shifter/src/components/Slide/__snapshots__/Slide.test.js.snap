// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide Component should render should show text and match snapshot 1`] = `
<li
  className="slide"
  tabIndex="0"
>
  <h1
    aria-label="slide 1"
    className="number-item"
  >
    1
  </h1>
</li>
`;
